Source/Modules/r.cxx

```diff
@@ -1052,9 +1052,10 @@ int R::OutputMemberReferenceMethod(String *className, int isSet,
     String *item = Getitem(el, j);
     String *dup = Getitem(el, j + 1);
     char *ptr = Char(dup);
-    ptr = &ptr[Len(dup) - 3];
+    int n = Len(dup);
+    ptr = &ptr[n - 3];
```

Comment:
If `n < 3` then this puts `ptr` before the start of the string, which is undefined behaviour if the string is at the start of the allocated buffer (I'm not sure how the DOH string stuff works, so maybe there's a count or something in front of the string data).

Either way, it would be cleaner to just move the setting of `ptr` inside the `n > 3` check.

Reply:
You are quite right - I thought about this and decided to be lazy as I'm planning to replace a lot of this code anyway. No excuses - I will send an update soon.